osu.Game/Screens/Edit/Timing/WaveformComparisonDisplay.cs

```diff
@@ -191,6 +191,10 @@ private void showFromTime(double time, bool animated)
 
         private void regenerateDisplay(bool animated)
         {
+            // Before a track is loaded, it won't have a valid length, which will break things.
+            if (!beatmap.Value.Track.IsLoaded)
+                return;
```

Comment:
This guard is fine and all to address the direct failure, but what are the guarantees the component will function correctly after hitting it? By which I mean: say that the waveform regeneration is requested, but the track isn't loaded yet. This early-guard will prevent the crash, but what is going to ensure that the waveform gets regenerated at some later point?

There _is_ this code in `Update()`:

https://github.com/ppy/osu/blob/0f7093cb22f6b658f561c5c77c0b8525605a5975/osu.Game/Screens/Edit/Timing/WaveformComparisonDisplay.cs#L168-L178

but the guarantees it gives come with a lot of caveats (component needs to be not hovered, not locked, and even then `showFromBeat()` / `showFromTime()` have early-return guards that prevent possibly-redundant regenerations from happening etc.)

I'd probably see this as a flow based around `Cached` or something. Everything that forces the display regeneration would instead invalidate the `Cached`, and `Update()` would check if the `Cached` is invalid and if it's not, run this method.

Reply:
Now that I think about it, this issue has actually been a thing for a while as far as I can tell. I've had cases the display doesn't update until I start to play the track (likely because it hit the case in this PR or something very similar but not as fatal).

Was hoping for a quick fix, but will look at doing this more correctly.

Reply:
I've used `Scheduler.AddOnce` as it feels like a pretty simple solution here.